docs/content/latest/api/ysql/extensions.md

```diff
@@ -0,0 +1,356 @@
+---
+title: Extensions
+linkTitle: Extensions
+description: Install and Use Extensions
+summary: Reference for YSQL Extensions
+menu:
+  latest:
+    identifier: api-ysql-extensions
+    parent: api-ysql
+    weight: 4400
+isTocNested: true
+showAsideToc: true
+---
+
+This page documents how to install and use YSQL-compatible PostgreSQL extensions.
+
+
+## Use Included Extensions
+
+These are extensions that are included into the standard YugaByteDB distribution and can be enabled in YSQL by simply running `CREATE EXTENSION`.
+
+### pgcrypto
+
+The pgcrypto extension provides various cryptographic functions.
+
+<!-- 
+  Use explicit html for example throughout this file to exclude example 
+  subsections from the generated table of contents.
+-->
+<h4> Example </h4>
+
+```sql
+CREATE EXTENSION pgcrypto;
+CREATE TABLE pgcrypto_example(id uuid PRIMARY KEY DEFAULT gen_random_uuid(), content text, digest text);
+INSERT INTO pgcrypto_example (content, digest) values ('abc', digest('abc', 'sha1'));
+SELECT * FROM pgcrypto_example;
+```
+
+```
+                  id                  | content |                   digest
+--------------------------------------+---------+--------------------------------------------
+ b8f2e2f7-0b8d-4d26-8902-fa4f5277869d | abc     | \xa9993e364706816aba3e25717850c26c9cd0d89d
+(1 row)
+```
+
+Read more about `pgcrypto` in the [PostgreSQL Docs](https://www.postgresql.org/docs/current/pgcrypto.html).
+
+
+### fuzzystrmatch
+
+The fuzzystrmatch extension provides several functions to determine similarities and distance between strings.
+
+<h4> Example </h4>
+
+```sql
+CREATE EXTENSION fuzzystrmatch;
+SELECT levenshtein('YugaByte', 'yugabyte'), metaphone('yugabyte', 8);
+```
+```
+ levenshtein | metaphone
+-------------+-----------
+           2 | YKBT
+(1 row)
+
+```
+
+### spi module
+
+The spi module includes several separate extensions using the Server Programming Interface (SPI) and triggers.
+The specific extensions currently supported in YSQL are:
+
+- `insert_username` Functions for Tracking Who Changed a Table
+- `moddatetime` Functions for Tracking Last Modification Time
+- `autoinc` Functions for Autoincrementing Fields
+- `refint` Functions for Implementing Referential Integrity
+
+<h4> Example </h4>
+
+- Set up a table with triggers for tracking modification time and user (role).
+  Connect with `ysqlsh` and run the commands below.
+  ```sql
+  CREATE EXTENSION insert_username;
+  CREATE EXTENSION moddatetime;
+
+  CREATE TABLE spi_test (
+      id int primary key,
+      content text,
+      username text not null,
+      moddate timestamp DEFAULT CURRENT_TIMESTAMP NOT NULL
+  );
+
+  CREATE TRIGGER insert_usernames
+    BEFORE INSERT OR UPDATE ON spi_test
+    FOR EACH ROW
+    EXECUTE PROCEDURE insert_username (username);
+
+  CREATE TRIGGER update_moddatetime
+    BEFORE UPDATE ON spi_test
+    FOR EACH ROW
+    EXECUTE PROCEDURE moddatetime (moddate);
+```
```

Comment:
Since this isn't indented properly, you get bad formatting all the way down.  Check the [processed file][processed-file].

[processed-file]: https://github.com/YugaByte/yugabyte-db/blob/e5ca530dc6fe818eac33ddbd14c5eda8e85c40ff/docs/content/latest/api/ysql/extensions.md